### PR TITLE
Feature/rs 134

### PR DESCRIPTION
**Pull request by aChanEP**
Description:

NOTE: A seperate branch has been created for this ticket and has been rebased on top of the latest monorepo changes that have been merged into master. Take a look here:  https://github.com/elasticpath/react-pwa-reference-storefront/pull/558 and close this pull request once everything has been tested and merged in the rebased version of this.

Linting:

- [ ] No linting errors

Component Updates:

- [ ] Component package requires update on npm [@elasticpath/store-components](https://www.npmjs.com/package/@elasticpath/store-components)

Tests:

- [ ] E2E tests (npm test run with `e2e`)
- [ ] Manual tests

Documentation:

- [ ] Requires documentation updates

### CODE REVIEW

File: app/src/containers/CheckoutPage.tsx [hunk 1]
```diff
@@ -574,13 +561,61 @@ class CheckoutPage extends React.Component<CheckoutPageProps, CheckoutPageState>
...
+          <label htmlFor="paymentMethod">
+            <div className="paymentMethodComponentRegion" data-region="paymentMethodComponentRegion" style={{ display: 'block' }}>
+              <PaymentMethodContainer displayName={payment} />
+            </div>
+          </label>
+        </div>
+        {deletable && (
+          <div className="payment-btn-cell">
+            <button className="ep-btn small checkout-delete-payment-btn" type="button" onClick={() => { this.handleDelete(payment.self.uri); }}>
+              Delete
```
Comment (shaunmaharaj) [inline app/src/containers/CheckoutPage.tsx]: hardcoded string

File: app/src/containers/CheckoutPage.tsx [hunk 2]
```diff
@@ -601,7 +636,12 @@ class CheckoutPage extends React.Component<CheckoutPageProps, CheckoutPageState>
                 </h2>
               </div>
               <div className="modal-body">
-                <PaymentFormMain onCloseModal={this.handleCloseNewPaymentModal} fetchData={this.fetchOrderData} />
+                {
+                  localStorage.getItem(`${Config.cortexApi.scope}_oAuthRole`) === 'PUBLIC' ? (
+                    <PaymentFormMain defaultPostSelection={false} showSaveToProfileOption={false} onCloseModal={this.handleCloseNewPaymentModal} fetchData={this.fetchOrderData} />)
+                    : (
+                      <PaymentFormMain defaultPostSelection={false} showSaveToProfileOption onCloseModal={this.handleCloseNewPaymentModal} fetchData={this.fetchOrderData} />)
```
Comment (shaunmaharaj) [inline app/src/containers/CheckoutPage.tsx]: The formatting here looks a bit off. For readability, can you please try formatting?

File: app/src/containers/ProfilePage.tsx [hunk 3]
```diff
@@ -181,12 +188,24 @@ class ProfilePage extends React.Component<RouteComponentProps, ProfilePageState>
...
   }
 
+  renderPayments() {
+    const { profileData } = this.state;
+
+    const disableAddPayment = false;
+
+    if (profileData._paymentmethods) {
+      if (profileData._paymentinstruments) {
+        return <ProfilePaymentMethodsMain paymentMethods={profileData._paymentmethods[0]} paymentInstruments={profileData._paymentinstruments[0]} onChange={this.fetchProfileData} disableAddPayment={disableAddPayment} />;
```
Comment (shaunmaharaj) [inline app/src/containers/ProfilePage.tsx]: excellent!!!!

File: components/src/CommonMockHttpResponses/registered_login_response.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+	"access_token": "9d97df81-e5cd-4a94-8edf-f2c8d1e15ec7",
+	"token_type": "bearer",
+	"expires_in": 604799,
+	"scope": "mobee",
+	"role": "REGISTERED",
+	"roles": ["REGISTERED"]
+}
```
Comment (shaunmaharaj) [inline components/src/CommonMockHttpResponses/registered_login_response.json]: missing newline char

File: components/src/PaymentForm/MockHttpResponses/GET/paymentInstrumentForm_response.json [hunk 5]
```diff
@@ -0,0 +1,746 @@
+{
+	"self": {
+		"type": "elasticpath.collections.links",
+		"uri": "/?zoom=defaultcart:order:paymentmethodinfo:element:paymentinstrumentform,defaultprofile:paymentmethods:element:paymentinstrumentform",
+		"href": "http://10.11.12.117:8080/cortex/?zoom=defaultcart:order:paymentmethodinfo:element:paymentinstrumentform,defaultprofile:paymentmethods:element:paymentinstrumentform"
```
Comment (shaunmaharaj) [inline components/src/PaymentForm/MockHttpResponses/GET/paymentInstrumentForm_response.json]: This is the internal dev environment. We'll need to update the url to point to the public instance `https://reference.epdemos.com`

File: components/src/PaymentForm/MockHttpResponses/GET/paymentInstrumentForm_response.json [hunk 6]
```diff
@@ -0,0 +1,746 @@
...
+							"PIC Field A": "",
+							"PIC Field B": "",
+							"display-name": ""
+						}
+					}]
+				}
+			]
+		}]
+	}]
+}
```
Comment (shaunmaharaj) [inline components/src/PaymentForm/MockHttpResponses/GET/paymentInstrumentForm_response.json]: missing new line char

File: components/src/PaymentForm/README.md [hunk 7]
```diff
@@ -2,7 +2,13 @@
...
+
+The checkbox labelled **Save This Payment Method To My Profile** will be available when setting **showSaveToProfileOption** to true and is also recommended to be set to true when consumed in a checkout flow.  It gives the user an option to save the payment information to the profile resource.
+
+Since this is a template component it is hardcoded to take the first **paymentinstrumentform** available in **paymentinstruments** resource.  When posting, arbitrary tokenized data is set to the respective **paymentinstrumentform** fields.  It is up to the implementor to make changes to how the component tokenizes data as this varies from solution to solution.  Comments in the code will guide as to which portions should be removed and filled.
```
Comment (shaunmaharaj) [inline components/src/PaymentForm/README.md]: Excellent documentation!

**Comment by shaunmaharaj** [comment]
@aChanEP Just to confirm, did you try running the E2E tests with this?

**Comment by aChanEP** [comment]
> @aChanEP Just to confirm, did you try running the E2E tests with this against 7.6?

No I have not.  Will set up local 7.6 env and test.

File: components/src/PaymentForm/MockHttpResponses/GET/paymentInstrumentForm_response.json (hunk 6 above)
Comment (aChanEP) [reply components/src/PaymentForm/MockHttpResponses/GET/paymentInstrumentForm_response.json]: wip

File: components/src/PaymentForm/README.md (hunk 7 above)
Comment (aChanEP) [reply components/src/PaymentForm/README.md]: thank you!

File: components/src/PaymentForm/MockHttpResponses/GET/paymentInstrumentForm_response.json (hunk 5 above)
Comment (aChanEP) [reply components/src/PaymentForm/MockHttpResponses/GET/paymentInstrumentForm_response.json]: WIP

File: components/src/CommonMockHttpResponses/registered_login_response.json (hunk 4 above)
Comment (aChanEP) [reply components/src/CommonMockHttpResponses/registered_login_response.json]: wip

File: app/src/containers/CheckoutPage.tsx (hunk 1 above)
Comment (aChanEP) [reply app/src/containers/CheckoutPage.tsx]: wip

File: app/src/containers/CheckoutPage.tsx (hunk 2 above)
Comment (aChanEP) [reply app/src/containers/CheckoutPage.tsx]: wip

**Pull request closed without merging**